### PR TITLE
Node/hex: bigint fixes

### LINKED ISSUE #1839: [Bug]: mismatch between Nodejs `bigint` types and (hex) `string` types
### Issue description

In the bindings, several interfaces are typed as BigInts, but expect or return hex encoded strings. This should be improved. Examples:
- TokenScheme: https://github.com/iotaledger/iota-sdk/blob/develop/bindings/nodejs/lib/types/block/output/token-scheme.ts#L39-L42
- CreateNativeTokenParams: https://github.com/iotaledger/iota-sdk/blob/develop/bindings/nodejs/lib/types/wallet/transaction-options.ts#L70
- There might be others. I haven't performed an exhaustive search

### Version

1.1.4

### Expected behaviour

Correct typings: if hex encoded amounts are returned then they should be typed as `HexEncodedAmount`, if it is of Javascript BigInt types, then they should be typed as `bigint`. Refrain from returning numbers, as that will result in lost precision.

### Actual behaviour

Incorrect typings for bigints. They return hexadecimal strings

### Can the issue reliably be reproduced?

Yes

### Steps to reproduce the issue

1. Create a new TokenScheme with circulating and total supply of `1000000000000000000000000000000n` BigInt type
2. Will result in a total supply of 1329227995784915872903807060280344576n, because it is being interpreted as a hex string

### Errors

_No response_

### Duplicate declaration

- [X] I have searched the issues tracker this issue and there is none

### PR DESCRIPTION
**Pull request by kwek20**
# Description of change

Fixes a few cases where hexstring wasnt parsed properly

**== Open question: ==**
Is there any way you can think of that makes this a breaking change?The original use still works: (It doesnt need `new INativeToken`)
```
let native: INativeToken = {
    id: "",
    amount: BigInt(1)
}
```

The alternative is changing the methods that use INativeToken to preparse like here:
https://github.com/iotaledger/iota-sdk/blob/2b44b886538b3c5dbbaf55a8ace38e8f44273bfe/bindings/nodejs/lib/wallet/account.ts#L550

## Links to any relevant issues

fixes #1839

## Type of change
- Bug fix (a non-breaking change which fixes an issue)

## How the change has been tested

Ran examples which touch the relevant code like list-outputs and create nativetoken

### CODE REVIEW

**Comment by kwek20** [comment]
@Tuditi Ive addressed the 2 cases you mention, and checked others. Do you think this is a breaking change or its fine?

**Comment by Tuditi** [comment]
It's fine, it can be seen as a fix. Based on the typings the developer would expect a BigInt.

**Review by DaughterOfMars — DISMISSED** [review]

**Review by Thoralf-M — CHANGES_REQUESTED** [review]
Changelog entry

File: bindings/nodejs/lib/types/models/native-token.ts [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
 // SPDX-License-Identifier: Apache-2.0
 
+import { Transform } from 'class-transformer';
 import type { HexEncodedString } from '../utils/hex-encoding';
+import { hexToBigInt } from '../utils/hex-encoding';
 /**
  * Native token.
  */
-export interface INativeToken {
+export class INativeToken {
```
Comment (Alex6323) [inline bindings/nodejs/lib/types/models/native-token.ts:10]: Should we rename this type then in 2.0 (remove the `I` prefix)?

Comment (kwek20) [reply bindings/nodejs/lib/types/models/native-token.ts:10]: yea, ill make at ticket

**Review by Thoralf-M — DISMISSED** [review]

**Review by DaughterOfMars — DISMISSED** [review]

**Review by Alex6323 — DISMISSED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

**Pull request merged**